src/librustc/ty/structural_impls.rs

```diff
@@ -847,27 +846,28 @@ impl<'tcx> TypeFoldable<'tcx> for ty::ProjectionPredicate<'tcx> {
 impl<'tcx> TypeFoldable<'tcx> for ty::ExistentialProjection<'tcx> {
     fn super_fold_with<'gcx: 'tcx, F: TypeFolder<'gcx, 'tcx>>(&self, folder: &mut F) -> Self {
         ty::ExistentialProjection {
-            trait_ref: self.trait_ref.fold_with(folder),
-            item_name: self.item_name,
             ty: self.ty.fold_with(folder),
+            substs: self.substs.fold_with(folder),
+            item_def_id: self.item_def_id,
         }
     }
 
     fn super_visit_with<V: TypeVisitor<'tcx>>(&self, visitor: &mut V) -> bool {
-        self.trait_ref.visit_with(visitor) || self.ty.visit_with(visitor)
+        self.substs.visit_with(visitor) || self.ty.visit_with(visitor)
     }
 }
 
 impl<'tcx> TypeFoldable<'tcx> for ty::ProjectionTy<'tcx> {
     fn super_fold_with<'gcx: 'tcx, F: TypeFolder<'gcx, 'tcx>>(&self, folder: &mut F) -> Self {
         ty::ProjectionTy {
-            trait_ref: self.trait_ref.fold_with(folder),
+            substs: self.substs.fold_with(folder),
             item_def_id: self.item_def_id,
         }
     }
 
     fn super_visit_with<V: TypeVisitor<'tcx>>(&self, visitor: &mut V) -> bool {
-        self.trait_ref.visit_with(visitor)
+        let ty::ProjectionTy { ref substs, item_def_id: _ } = *self;
```

Comment:
I don't know why @nikomatsakis mentioned this style when *everything else* does:
```rust
self.substs.visit_with(visitor)
```
if we want uniformity we'll be auto-deriving these at some point.